src/net_processing.cpp

```diff
@@ -2744,7 +2753,13 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // set nodes not capable of serving the complete blockchain history as "limited nodes"
         pfrom.m_limited_node = (!(nServices & NODE_NETWORK) && (nServices & NODE_NETWORK_LIMITED));
 
-        if (auto tx_relay = peer->GetTxRelay(); tx_relay != nullptr) {
+        // We only initialize the m_tx_relay data structure if:
+        // - this isn't an outbound block-relay-only connection; and
+        // - fRelay=true or we're offering NODE_BLOOM to this peer
+        //   (NODE_BLOOM means that the peer may turn on tx relay later)
+        if (!pfrom.IsBlockOnlyConn() &&
```

Comment:
I agree that we could skip it, and also agree that it's out of scope for this PR (which shouldn't change behaviour).

Also note that we only have at most one feeler connection at a time, so this wouldn't result in much resource saving (whereas it's potentially very significant for inbound block-relay-only peers, since there could be many of them).